app/src/main/java/com/seafile/seadroid2/monitor/SeafileObserver.java

```diff
@@ -125,8 +126,11 @@ public void onFileChange(File file) {
         Log.d(DEBUG_TAG, path + " was modified!");
         SeafCachedFile cachedFile = watchedFiles.get(path);
         if (cachedFile != null) {
-            //  file  size  is change
-            if (file.length() == cachedFile.fileOriginalSize) {
+            //  office or txt file need to update file
+            String fileMime = FileMimeUtils.getFileMime(file);
+            boolean needUpdateFile = FileMimeUtils.isOfficeOrTextFile(fileMime);
+            //  file  size  is change  ,
+            if (file.length() == cachedFile.fileOriginalSize && !needUpdateFile) {
```

Comment:
This does not work in a generic way (the mime type check). A really bad idea from my point of view.

It is not more than a dirty hack.